src/Features/CSharp/Portable/UsePatternMatching/CSharpAsAndNullCheckCodeFixProvider.cs

```diff
@@ -77,12 +77,20 @@ private static ExpressionSyntax GetCondition(SyntaxNode node)
             var conditionPart = (BinaryExpressionSyntax)conditionLocation.FindNode(cancellationToken);
             var asExpression = (BinaryExpressionSyntax)asExpressionLocation.FindNode(cancellationToken);
 
-            var updatedConditionPart = SyntaxFactory.IsPatternExpression(
+            var updatedConditionPart = (ExpressionSyntax)SyntaxFactory.IsPatternExpression(
```

Comment:
❗️ Use an explicitly-typed variable rather than a cast. This ensures a compilation error occurs for any case where an implicit cast is not available (which in theory ensures the operation is a widening conversion, though we have cases even in this codebase where that is not the case).